torch/_dynamo/utils.py

```diff
@@ -761,10 +762,24 @@ def clone_inputs(example_inputs):
     return res
 
 
+def skip_frame_if_in_functorch_mode(val: torch.Tensor):
+    try:
+        val.data_ptr()  # will throw for functorch tensors
+    except RuntimeError as e:
+        from .exc import SkipFrame
+
+        # This will be GradTrackingTensor/BatchedTensor/etc
+        functorch_subclass_name = re.sub(r"\(.*", "", repr(val))
```

Comment:
@jansel, can you make this less functorch-centric? functorch is not the only place where one can get a Tensor with no data_ptr.

Reply:
This should work for all of them.  Though the functorch ones are the only ones that show up in tests.